receiver/podmanreceiver/receiver.go

```diff
@@ -29,40 +29,48 @@ type metricsReceiver struct {
 	clientFactory clientFactory
 	scraper       *ContainerScraper
 	mb            *metadata.MetricsBuilder
+	cancel        context.CancelFunc
 }
 
 func newMetricsReceiver(
-	_ context.Context,
 	set receiver.CreateSettings,
 	config *Config,
-	nextConsumer consumer.Metrics,
 	clientFactory clientFactory,
-) (receiver.Metrics, error) {
-	err := config.Validate()
-	if err != nil {
-		return nil, err
-	}
-
+) *metricsReceiver {
 	if clientFactory == nil {
 		clientFactory = newLibpodClient
 	}
 
-	recv := &metricsReceiver{
+	return &metricsReceiver{
 		config:        config,
 		clientFactory: clientFactory,
 		set:           set,
 		mb:            metadata.NewMetricsBuilder(config.MetricsBuilderConfig, set),
 	}
+}
 
-	scrp, err := scraperhelper.NewScraper(metadata.Type.String(), recv.scrape, scraperhelper.WithStart(recv.start))
+func createMetricsReceiver(
+	_ context.Context,
+	params receiver.CreateSettings,
+	config component.Config,
+	consumer consumer.Metrics,
+) (receiver.Metrics, error) {
+	podmanConfig := config.(*Config)
+
+	recv := newMetricsReceiver(params, podmanConfig, nil)
+	scrp, err := scraperhelper.NewScraper(metadata.Type.String(), recv.scrape, scraperhelper.WithStart(recv.start), scraperhelper.WithShutdown(recv.shutdown))
 	if err != nil {
 		return nil, err
 	}
-	return scraperhelper.NewScraperControllerReceiver(&recv.config.ControllerConfig, set, nextConsumer, scraperhelper.AddScraper(scrp))
+	return scraperhelper.NewScraperControllerReceiver(&recv.config.ControllerConfig, params, consumer, scraperhelper.AddScraper(scrp))
 }
 
 func (r *metricsReceiver) start(ctx context.Context, _ component.Host) error {
-	var err error
+	err := r.config.Validate()
```

Comment:
Why are you calling `Config.Validate` in the component start function? I don't think this function should be called by component authors. It is called for each component during collector setup:
- [otelcol.Collector Run](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.101.0/otelcol/collector.go#L287)
  - [otelcol.Collector setupConfigurationComponents](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.101.0/otelcol/collector.go#L196)
    - [otelcol.Config Validate](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.101.0/otelcol/config.go#L45)